server/spec/api/v1/stacks_spec.rb

```diff
@@ -151,6 +151,14 @@
         expect(json_response['parent']).to match hash_including('name' => stack.name, 'id' => "#{stack.grid.name}/#{stack.name}")
       end
 
+      context 'when parent stack does not exist' do
+        it 'returns stack json with parent name excluding parent id' do
+          child_stack_1.destroy
+          get "/v1/stacks/#{child_stack_1_2.to_path}", nil, request_headers
+          expect(json_response['parent']).to match hash_including('name' => stack.name, 'id' => nil)
```

Comment:
I suggest using explicit `'name' => 'child-stack-1` matches in specs when reasonable, rather than trying to re-use the `let` variables like this.

```
  1) /v1/stacks GET /:name nested stacks when parent stack does not exist returns stack json with parent name excluding parent id
     Failure/Error: expect(json_response['parent']).to match hash_including('name' => stack.name, 'id' => nil)
       expected {"id"=>nil, "name"=>"child-stack-1"} to match #<RSpec::Mocks::ArgumentMatchers::HashIncludingMatcher:0x00000004d190d0 @expected={"name"=>"stack", "id"=>nil}>
       Diff:
       @@ -1,3 +1,3 @@
       -#<RSpec::Mocks::ArgumentMatchers::HashIncludingMatcher:0x00000004d190d0
       - @expected={"name"=>"stack", "id"=>nil}>
       +"id" => nil,
       +"name" => "child-stack-1",
       
     # ./spec/api/v1/stacks_spec.rb:158:in `block (5 levels) in <top (required)>'
```